backend/src/worker.rs

```diff
@@ -89,6 +89,13 @@ pub async fn run_worker(
 
     insert_initial_ping(worker_instance, &worker_name, ip, db).await;
 
+    let job_duration_seconds = prometheus::register_histogram!(prometheus::HistogramOpts::new(
+        "job_duration_seconds",
+        "The duration between receiving a job and completing it"
+    )
+    .const_label("name", &worker_name))
+    .expect("register prometheus histogram");
+
```

Comment:
thoughts about adding a counter for reporting the number of jobs that failed ?

We can also get uptime by reporting the boot time of the worker. And I would add a label based on the language of the job: python or deno. And a label for the workspace_id.

Much less important: we can also do prometheus for the api server part: An histogram for the duration of each request.

Reply:
> thoughts about adding a counter for reporting the number of jobs that failed ?

Added but it reports failures that bubble up to `run_worker`, not jobs that were unsuccessful. Not 100% sure if this is what you wanted.

Reply:
> Added but it reports failures that bubble up to run_worker, not jobs that were unsuccessful. Not 100% sure if this is what you wanted. 

That's not great because that error handler is actually a "last resort" error handler and hopefully most errors and especially errors from within jobs are caught earlier (you will see there is another error handler deeper into the handlers).

The perfect place to increase that counter is within: `add_completed_job_error` but it means we have to pass that counter everywhere which is inconvenient. lmk if you have any ideas on how to improve this.

Reply:
I _think_ we can use [tokio::task::LocalKey](https://docs.rs/tokio/1.20.1/tokio/task/struct.LocalKey.html#method.with) to make it like a "thread" local thing. The code is still highly coupled but refactoring it isn't trivial either.